tests/testthat/test-helper_functions.R

```diff
@@ -0,0 +1,14 @@
+test_that("check_and_prepare_by_group() returns expected value", {
+  expect_identical(check_and_prepare_by_group(by_group = NULL), NULL)
+  expect_identical(check_and_prepare_by_group(by_group = "NULL"), NULL)
+  expect_identical(check_and_prepare_by_group(by_group = "foo"), "foo")
+  expect_identical(check_and_prepare_by_group(by_group = "foo, bar"), c("foo", "bar"))
```

Comment:
nit: is there any possibility of `by_group` having > length 1? I think it should still work, but could maybe test what happens with `check_and_prepare_by_group(by_group = c("foo", "bar")`

Reply:
there is not, unless you play around with `by_group` after reading the `config.yml`. But any string read from the `yml` using `config::get()` is returned as a vector of length one in my understanding

Reply:
in that sense I would prefer adding `stop_if_not_length(by_group, 1)`